pandas/io/sas/sas7bdat.py

```diff
@@ -687,6 +689,39 @@ def _read_next_page(self):
 
         return False
 
+    @staticmethod
+    def _convert_dates(sas_dates):
+        try:
+            return pd.to_datetime(sas_dates, unit="d", origin="1960-01-01")
+        except OutOfBoundsDatetime:
+            # convert to datetime.datetime rather than np.datetime64
+            # NB generally better support in pandas for datetime than date
+            warn(
+                "SAS date > pandas.Timestamp.max, returning "
+                "datetime.datetime objects instead"
+            )
+            return sas_dates.apply(
+                lambda sas_date_float: datetime(1960, 1, 1)
+                + timedelta(days=sas_date_float)
+            )
+
+    @staticmethod
+    def _convert_datetimes(sas_datetimes):
+        try:
+            return pd.to_datetime(sas_datetimes, unit="s", origin="1960-01-01")
+        except OutOfBoundsDatetime:
+            # convert to datetime.datetime rather than np.datetime64
+            # SAS float64 lacks precision for more than ms resolution so the
+            # fit to datetime.datetime is ok
+            warn(
+                "SAS datetime > pandas.Timestamp.max, returning "
```

Comment:
is there a reason to have 2 very similar routines here?

Reply:
Hi,  I've combined the two into a single function